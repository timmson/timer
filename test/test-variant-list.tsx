import React from "react"
import {render, screen} from "@testing-library/react"
import user from "@testing-library/user-event"
import "@testing-library/jest-dom"
import "regenerator-runtime/runtime"

import VariantList from "../src/variant-list"
import Context from "../src/context"

describe("VariantList should", () => {

	const expectedAction = {type: "ACTION_SET_TIME", value: "01:00"}

	test("give possibility to click", async () => {
		render(
			<Context.Provider value={(action) => expect(action).toEqual(expectedAction)}>
				<VariantList variants={[1, 2]}/>
			</Context.Provider>
		)

		expect(screen.getByText(/02:00/)).toBeInTheDocument()
		await user.click(screen.getByText(/01:00/))

		expect.assertions(2)

	})
})
