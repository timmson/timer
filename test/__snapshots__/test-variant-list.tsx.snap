// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VariantList should return variants 1`] = `
<div
  className="row"
>
  <div>
    01:00
  </div>
  <div>
    02:00
  </div>
</div>
`;
